crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -528,11 +528,11 @@ mod tests {
         let opts = ResolverOpts::default();
         let ns_config = { tcp };
         let name_server = NameServer::new_with_provider(ns_config, opts, conn_provider.clone());
-        let name_servers = Arc::new(vec![name_server]);
+        let name_servers: Arc<[_]> = Arc::from([name_server]);
```

Comment:
Do we need this type annotation here? Might be a little nicer to write it as `Arc::<[_]>::from()` (which prevents repeating the `Arc` part at least.

Reply:
The compiler gets confused on the from and to type, slice vs array.

Reply:
yeah, can't get rid of this.